runtime/compiler/control/HookedByTheJit.cpp

```diff
@@ -5928,6 +5928,84 @@ class CompilationDensity
       }
    }; // class CompilationDensity
 
+#if defined(J9VM_OPT_CRIU_SUPPORT)
+static void suspendSamplerThreadForCheckpoint(J9VMThread *samplerThread, J9JITConfig *jitConfig, TR::CompilationInfo *compInfo)
+   {
+   compInfo->acquireCompMonitor(samplerThread);
+   if (compInfo->shouldSuspendThreadsForCheckpoint())
+      {
+      PORT_ACCESS_FROM_JITCONFIG(jitConfig);
+
+      // Must acquire this with the comp monitor in hand to ensure
+      // consistency with the checkpointing thread.
+      j9thread_monitor_enter(jitConfig->samplerMonitor);
+
+      // Because this thread has the comp monitor in hand, and because
+      // shouldSuspendThreadsForCheckpoint() returned true, it is not possible
+      // for the Sampler Thread to have any other state other than SAMPLE_THR_INITIALIZED
+      TR_ASSERT_FATAL(compInfo->getSamplingThreadLifetimeState() == TR::CompilationInfo::SAMPLE_THR_INITIALIZED,
```

Comment:
No, because in all cases related to shutdown, we call `stopCompilationThreads()` before `JitShutdown`; in `stopCompilationThreads` we call `interruptCheckpoint()` which would result in `compInfo->shouldSuspendThreadsForCheckpoint()` returning false, and thus we'd never get into the process of suspending the sampler thread for checkpoint. On the other hand, once we reach this point, because the sampler thread has the comp monitor in hand, it is not possible for the shutdown thread to call `interruptCheckpoint()` because it can only do so with the comp monitor in hand.